client/src/dateUtils.ts

```diff
@@ -55,6 +58,16 @@ export function dateToQuery(queryKey, value) {
       [startKey]: startDateStart,
       [endKey]: startDateEnd
     }
+  } else if (value.relative === LAST_X_DAYS) {
+    const days = value.days
+    if (days && !isNaN(days)) {
```

Comment:
This discards `0` as a valid number; suggest to change to `if (utils.isNumeric(days)) {`